packages/features/bookings/Booker/components/DatePicker.tsx

```diff
@@ -35,6 +36,21 @@ export const DatePicker = ({
     shallow
   );
   const nonEmptyScheduleDays = useNonEmptyScheduleDays(schedule?.data?.slots);
+  const browsingDate = month ? dayjs(month) : dayjs().startOf("month");
+
+  const onMonthChange = (date: Dayjs) => {
+    setMonth(date.format("YYYY-MM"));
+    setSelectedDate(date.format("YYYY-MM-DD"));
+    setDayCount(null); // Whenever the month is changed, we nullify getting X days
+  };
+
+  useEffect(() => {
+    const currentMonth = dayjs().startOf("month").format("YYYY-MM");
+    const browsingMonth = browsingDate.format("YYYY-MM");
+    if (currentMonth === browsingMonth && nonEmptyScheduleDays.length === 0) {
```

Comment:
Considering `nonEmptyScheduleDays` work on schedule?.data?.slots which could be undefined and could possibly be undefined initially and then later set when the data is updated, I think we should confirm this scenario and accordingly ensure that we consider nonEmptyScheduleDays only if it has valid value.

I want to avoid the scenario where we accidentally move to next month in a case where slots haven't loaded.